File: 851-goat-latin/goat-latin.go
func toGoatLatin(sentence string) string {
    words := strings.Fields(sentence)
    vowels := "aeiouAEIOU"

    for i, word := range words {
        if strings.ContainsRune(vowels, rune(word[0])) {
            words[i] = word + "ma"
        } else {
            words[i] = word[1:] + string(word[0]) + "ma"
        }

        words[i] = words[i] + strings.Repeat("a", i+1)
    }

    return strings.Join(words, " ")
}

File: 851-goat-latin/goat-latin.py
class Solution:
    def toGoatLatin(self, sentence: str) -> str:
        vowels = set("aeiouAEIOU")
        res = []
        
        for i, word in enumerate(sentence.split(), 1):
            if word[0] in vowels:
                new_word = word + "ma"
            else:
                new_word = word[1:] + word[0] + "ma"
            
            new_word += ("a" * i)
            res.append(new_word)
            
        return " ".join(res)